projects/gnutls/Dockerfile

```diff
@@ -0,0 +1,31 @@
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+FROM ossfuzz/base-libfuzzer
+MAINTAINER alex.gaynor@gmail.com
+RUN apt-get install -y make autoconf automake libtool autopoint libnettle6 nettle-dev pkg-config gperf bison autogen texinfo curl
+
+RUN git clone https://gitlab.com/gnutls/gnutls.git
+RUN cd gnutls && git submodule update --init
+
+# Using the client_corpus transcripts from boringssl, they're a decent starting point.
```

Comment:
So one of the things I did recently is to make the random number generator deterministic. I only need this during the client and server fuzzer so far. I'm hoping to be able to get past all the checks that way, but I'm not sure I'm currently getting past them. I might either need to help the fuzzer to find the right value or do like you and modify the code to still do the checks but then continue if it failed anyway.

It really helps some fuzzers if you can give it interesting input. If it was stuck, adding a new interesting file will will suddenly make it find lots of new stuff. So it's really important that you find a good initial set, and it's not always easy to find or generate it.

For the x509 fuzzer what really helped was adding lots of real certificates. That really boosted the coverage. I imported 250K real certificates it seems. For the crl fuzzer I imported all CA CRLs I could download which is like 25K files. But for something like cms it seems hard to find things.

I actually have no idea how good the client and server corpus is currently. I planned on dumping files from the test suite like you seem to be doing, but I never got around of doing that.

In any case, I try to update the corpus in openssl's repository with the latest minimized coverage regularly.